### DISCUSSION TITLE
Need help: where can I find the explanation of DSL grammar in detail?

### DISCUSSION THREAD

**Discussion by hunter2009pf** [langgenius/dify#8090] [Help] [answered]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

Yes. I want to define my own dsl file. But I don't know its grammar and syntax.

**Comment by laipz8200** [comment]
Dify's DSL is essentially the JSON structure of the data in the front-end canvas. You can understand it by reading the code.

**Reply by hunter2009pf** [reply]
> Dify's DSL is essentially the JSON structure of the data in the front-end canvas. You can understand it by reading the code.

If it is constructed as json, why on dify platform we need to import a yaml file as DSL? Am I miss-understanding something or there is a conversion between yaml DSL and json DSL?

**Reply by laipz8200** [reply, marked as answer]
This is just a design consideration. In practice, the JSON passed from the front end is converted into a Python dictionary or model on the back end. These objects are then converted into YAML, but they basically retain their original structure.

**Reply by hunter2009pf** [reply]
> This is just a design consideration. In practice, the JSON passed from the front end is converted into a Python dictionary or model on the back end. These objects are then converted into YAML, but they basically retain their original structure.

Answer my question definitely. By the way, do you have any wechat group that I can join? I am freshman to Dify and I really want to discuss technical points with you on time. Thanks a lot~

**Reply by laipz8200** [reply]
@crazywoola
